lib/punycode.js

```diff
@@ -194,6 +194,9 @@ const adapt = function(delta, numPoints, firstTime) {
  * @returns {String} The resulting string of Unicode symbols.
  */
 const decode = function(input) {
+  if (!(typeof input === 'string' || Array.isArray(input) && input.length === 0)) {
+    throw new TypeError('Argument must be a string or an empty array');
```

Comment:
Sorry for not being clear, but why are _any_ arrays allowed? The documentation says only strings are allowed.